### PR TITLE
run tvmc with temp working dir; fix model group lookup

### PR DESCRIPTION
**Pull request by rafzi**

### CODE REVIEW

**Comment by PhilippvK** [comment]
@rafzi can you shortly explain what was wrong about the model group handling? I thought I fixed it last week.

File: mlonmcu/models/lookup.py [hunk 1]
```diff
@@ -310,12 +310,16 @@ def apply_modelgroups(models, context=None):
...
     for directory in directories:
-        groups = list_modelgroups(directory)
-        for i, group in enumerate(groups):
-            if group.name in models:
-                models = [*models[:i], *group.models, *models[i + 1 :]]
-                break
+        for group in list_modelgroups(directory):
+            if group.name in groups and groups[group.name] != group.models:
+                raise RuntimeError(
+                    f"The model group '{group.name}' has conflicting definitions. Used the following directories: {directories}"
```
Comment (PhilippvK) [inline mlonmcu/models/lookup.py:318]: Convert into multi-line string to fix lint issue

**Review by PhilippvK — CHANGES_REQUESTED** [review]

**Comment by rafzi** [comment]
The index `i` for the list replacement didn't really make sense as that was referring to the index of the group list, not the model list.

For the linting, I found out that I have to add `--experimental-string-processing` to black to make it format that line.

**Comment by PhilippvK** [comment]
@rafzi thanks. Make sense to me. I will try out the model group handling tomorrow and try to add a unit test which would have catched this issue.

**Pull request merged**